src/psyclone/tests/nemo/transformations/openacc/loop_directive_test.py

```diff
@@ -165,6 +166,48 @@ def test_seq_loop(parser):
             "  do ji = 1, jpj, 1\n" in code)
 
 
+def test_gang_loop(parser):
```

Comment:
Thanks for adding these tests. It turns out that this file is a bit old and we have improved things since these tests were written. For instance, you can use the `fortran_reader` fixture:

    def test_gang_loop(fortran_reader):
        psyir = fortran_reader.psyir_from_source(SINGLE_LOOP)
        schedule = psyir.children...[something]

You can probably also reduce duplication by using pytest.parametrize (sic):

    @pytest.mark.parametrize("name", ["gang", "vector"])
    def test_loop_clauses(fortran_reader, clause):
        ...
        replace use of gang/vector with `clause`

Reply:
Sorry @arporter, please could you give a bit more information? I tried
```
@@ -166,13 +166,11 @@ def test_seq_loop(parser):
             "  do ji = 1, jpj, 1\n" in code)
 
 
-def test_gang_loop(parser):
+def test_gang_loop(fortran_reader):
     ''' Check that we can apply the transformation with the 'gang'
     clause. '''
-    reader = FortranStringReader(SINGLE_LOOP)
-    code = parser(reader)
-    psy = PSyFactory(API, distributed_memory=False).create(code)
-    schedule = psy.invokes.invoke_list[0].schedule
+    psyir = fortran_reader.psyir_from_source(SINGLE_LOOP)
+    schedule = psyir.children[0]
     acc_trans = TransInfo().get_trans_name('ACCLoopTrans')
     # An ACC Loop must be within a KERNELS or PARALLEL region
     kernels_trans = TransInfo().get_trans_name('ACCKernelsTrans')
```
and similar for `vector`, but got the traceback
```
        # Check that the front-end is valid
        sched = node_list[0].ancestor((NemoInvokeSchedule, DynInvokeSchedule))
        if not sched:
            raise NotImplementedError(
>               "OpenACC kernels regions are currently only supported for the "
                "nemo and dynamo0.3 front-ends")
E           NotImplementedError: OpenACC kernels regions are currently only supported for the nemo and dynamo0.3 front-ends

src/psyclone/transformations.py:2547: NotImplementedError
```

Is there a way that I can tell the `fortran_reader` that I'm using the NEMO API? I couldn't see anything by scanning the source code.

Reply:
Ah! That's a pain. That check is overly restrictive but hey ho. You'll just have to ignore my first suggestion, sorry.

Reply:
Okay, no problem. See 0b2720e0a9dd488b8247d2981709a121859975ec for the requested parametrisation change.